contents/tutorials/chatgpt-analytics.md

```diff
@@ -0,0 +1,318 @@
+Additionally, you can also breakdown your costs by model. To do this, click **+ Add breakdown** and select `model` from the event properties list.
+
+<ProductScreenshot
+  imageLight={TotalCostLight} 
+  imageDark={TotalCostDark} 
+  alt="Total ChatGPT model costs in PostHog" 
+  classes="rounded"
+/>
+
+### Average cost per user
+
+This metric helps give you an idea of how your costs will scale as your product grows. Creating this insight is similar to creating the one above, however we use **formula mode** to divide the total cost by the total number of users:
+
+1. Set the event to `chat_completion`
+2. Click on **Total count** to show a dropdown. Click on **Property value (sum)**.
+3. Select the `total_cost_in_dollars` property.
+4. Click **+ Add graph series**.
+5. Keep the event name as `All events`, but change the value from `Total count` to `Unique users`.
+6. Click **Enable formula mode**.
+7. In the formula box, enter `A/B`.
+
+Once again, note that it may show `0` if the number is smaller than `0.01`.
+
+<ProductScreenshot
+  imageLight={CostPerUserLight} 
+  imageDark={CostPerUserDark} 
+  alt="Total ChatGPT costs per user in PostHog" 
+  classes="rounded"
+/>
+
+### Average API response time
+
+ChatGPT's API response time can take long, especially for longer outputs, so it's useful to keep an eye on this. To do this, first we need to modify our event capturing to also include the response time:
+
+```js file=App.js
+const fetchChatGPTResponse = async () => {
+    try {
+
+      // your existing code...
+
+      const startTime = performance.now(); 
+      const chatCompletion = await openai.chat.completions.create({
+        messages: [{ 
+          role: "user",
+          content: userInput }],
+        model: selectedModel.name,
+      });
+      const endTime = performance.now();
+      const responseTime = endTime - startTime;
+
+      posthog.capture('chat_completion', {
+        model: chatCompletion.model,
+        prompt: userInput,
+        prompt_tokens: chatCompletion.usage.prompt_tokens,
+        completion_tokens: chatCompletion.usage.completion_tokens,
+        total_tokens: chatCompletion.usage.total_tokens,
+        input_cost_in_dollars: chatCompletion.usage.prompt_tokens * selectedModel.token_input_cost,
+        output_cost_in_dollars: chatCompletion.usage.prompt_tokens * selectedModel.token_output_cost,
+        total_cost_in_dollars: inputCostInDollars + outputCostInDollars,
+        response_time_in_ms: responseTime
+      })
+
+      // your existing code...
+```
+
+Then, after capturing a few events, create a new insight to calculate the average response time:
+
+1. Set the event to `chat_completion`
+2. Click on **Total count** to show a dropdown. Click on **Property value (average)**.
+3. Select the `response_time_in_ms` property.
+
+<ProductScreenshot
+  imageLight={ResponseTimeLight} 
+  imageDark={ResponseTimeDark} 
+  alt="Average API response time in PostHog" 
+  classes="rounded"
+/>
+
+## Next steps
```

Comment:
Not sure if you need this section tbh, feels like you've covered all of this :)

Reply:
Cool. Felt like I wanted to include some other examples of metrics to look at